puree/src/main/java/com/mercari/puree/TagPattern.java

```diff
@@ -0,0 +1,96 @@
+package com.mercari.puree;
+
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+public class TagPattern {
+
+    private static final String SEPARATOR = "\\.";
+    private static final String ALL_WILD_CARD = "**";
+    private static final String WILD_CARD = "*";
+    private static final Pattern VALID_PATTERN_ALPHANUMERIC = Pattern.compile("^[A-Za-z0-9]+");
+
+    private String pattern;
+
+    public TagPattern() {
```

Comment:
Do we really need an empty constructor?

Reply:
As talked offline, exposes this default behavior by a static method (like TagPattern.matchAll()) is preferable